vaadin-spreadsheet-flow-parent/vaadin-spreadsheet-flow-client/src/main/java/com/vaadin/addon/spreadsheet/client/PopupButtonServerRpcImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.vaadin.addon.spreadsheet.client;
+
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.vaadin.component.spreadsheet.client.js.SpreadsheetServerRpcImpl;
+import com.vaadin.component.spreadsheet.client.js.SpreadsheetServerRpcImpl.JsConsumer;
+
+@SuppressWarnings("serial")
+public class PopupButtonServerRpcImpl implements PopupButtonServerRpc {
+
+    private JsConsumer<Void> popupButtonClickCallback;
+    private JsConsumer<Void> popupCloseCallback;
+
+    @Override
+    public void onPopupButtonClick(int row, int column) {
+        call(popupButtonClickCallback, row, column);
+    }
+
+    @Override
+    public void onPopupClose(int row, int column) {
+        call(popupCloseCallback, row, column);
+    }
+
+    public void setPopupCloseCallback(JsConsumer<Void> popupCloseCallback) {
+        this.popupCloseCallback = popupCloseCallback;
+    }
+
+    public void setPopupButtonClickCallback(
+            JsConsumer<Void> popupButtonClickCallback) {
+        this.popupButtonClickCallback = popupButtonClickCallback;
+    }
+
+    private native void call(JsConsumer<?> fnc, Object... args) /*-{
```

Comment:
Please don't duplicate code and reuse from the `SpreadsheetServerRpcImpl`

Reply:
Decided to make `SpreadsheetServerRpcImpl` implement `PopupButtonServerRpc`, since I was creating only one instance per spreadsheet anyway.